### PR TITLE
MythicBlockerFeature: Make check stricter by checking item type (and util housekeeping)

### PR DESCRIPTION
**Pull request by kristofbolyai**
Fixes the issue where you cannot close a chest with a 2 start ingredient inside.

### CODE REVIEW

File: common/src/main/java/com/wynntils/mc/utils/ItemUtils.java [hunk 1]
```diff
@@ -185,6 +187,20 @@ public static void removeItemLore(List<Component> tooltip) {
         tooltip.removeAll(toRemove);
     }
 
+    /**
+     * Returns true if the passed item is a Wynncraft item (armor, weapon, accessory)
+     */
+    public static boolean isWynnItem(ItemStack itemStack) {
```
Comment (P0keDev) [inline common/src/main/java/com/wynntils/mc/utils/ItemUtils.java]: these two new functions probably belong in WynnItemUtils, as does the original `isMythic` one - the minecraft package is meant for stuff that isn't specific to wynncraft

**Review by P0keDev — COMMENTED** [review]
the danger of checking for custom itemstacks like this is that it wouldn't prevent the closing of a chest if it contained an identified mythic that was new and hadn't been added to the api. wouldn't be an issue with mythic boxes, though. i think it would be safer to keep the box check like this, but replace the gear check with a more basic check for the purple "Mythic Item" line in the lore (i think that's what we look for in 1.12?)

**Comment by kristofbolyai** [comment]
> the danger of checking for custom itemstacks like this is that it wouldn't prevent the closing of a chest if it contained an identified mythic that was new and hadn't been added to the api. wouldn't be an issue with mythic boxes, though. i think it would be safer to keep the box check like this, but replace the gear check with a more basic check for the purple "Mythic Item" line in the lore (i think that's what we look for in 1.12?)

I actually did that before I switched to instance based stuff.

File: common/src/main/java/com/wynntils/mc/utils/ItemUtils.java (hunk 1 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/mc/utils/ItemUtils.java]: WynnItemMatcher exists, we should use that.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/mc/utils/ItemUtils.java]: Fixed this.

File: common/src/main/java/com/wynntils/mc/utils/ItemUtils.java [hunk 2]
```diff
@@ -26,7 +25,7 @@
 
 public class ItemUtils {
 
-    private static final Pattern ITEM_RARITY_PATTERN =
+    public static final Pattern ITEM_RARITY_PATTERN =
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/mc/utils/ItemUtils.java]: Actually not too sure about this, should this be public or should this be re-defined in the other file? Should we perhaps have a file containing these patterns?

Comment (P0keDev) [reply common/src/main/java/com/wynntils/mc/utils/ItemUtils.java]: honestly i don't think this pattern should exist in this file at all - it's used in a method that removes wynn flavor text, which should go in `WynnItemUtils` (and also be renamed to something more clear). it's fine to keep the pattern public once it's moved imo

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/mc/utils/ItemUtils.java]: Moved the method. About the pattern file, let's open a discussion and see what others think.

**Review by P0keDev — APPROVED** [review]
i'm fine with pushing this as is and resolving the pattern stuff in a later PR

**Review by magicus — APPROVED** [review]

File: common/src/main/java/com/wynntils/wc/utils/WynnItemMatchers.java [hunk 3]
```diff
@@ -88,4 +92,21 @@ public static boolean isCosmetic(ItemStack stack) {
         }
         return false;
     }
+
+    public static boolean isMythic(ItemStack stack) {
+        return stack.getDisplayName().getString().contains(ChatFormatting.DARK_PURPLE.toString());
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/wc/utils/WynnItemMatchers.java:97]: IMO the check for whether its an item and unidentified should be done in this method as well

**Pull request merged**